lib/mobility-core/src/Kernel/External/Maps/Interface/Google.hs

```diff
@@ -94,6 +96,53 @@ getDistances cfg GetDistancesReq {..} = do
     splitListByAPICap inputList = do
       List.chunksOf 25 $ toList inputList
 
+getDistance ::
+  ( EncFlow m r,
+    CoreMetrics m,
+    HasCoordinates a,
+    HasCoordinates b
+  ) =>
+  GoogleCfg ->
+  GetDistanceReq a b ->
+  m (GetDistanceResp a b)
+getDistance cfg GetDistanceReq {..} = do
+  let googleMapsUrl = cfg.googleMapsUrl
+  key <- decrypt cfg.googleKey
+  let placeOrigin = latLongToPlace $ getCoordinates origin
+      placeDestination = latLongToPlace $ getCoordinates destination
+  response <- mkDirectionsCall $ GoogleMaps.directions googleMapsUrl key placeOrigin placeDestination mode Nothing
+  let routesData = map (sumLegsDistancesAndDuration . (.legs)) response.routes
+  case nonEmpty routesData of
+    Nothing -> throwError (InternalError "Empty routes list in Distance API.")
+    Just routesDataNE -> do
+      let deciderThreshold = cfg.routeDurationDeciderThreshold
+      let (distance, duration) = dataDecider deciderThreshold routesDataNE
+      pure $ GetDistanceResp {status = response.status, ..}
+  where
+    sumLegsDistancesAndDuration =
+      foldr
+        (\x (accDistance, accDuration) -> (Meters x.distance.value + accDistance, Seconds x.duration.value + accDuration))
+        (0, 0)
+    mode = mapToMode <$> travelMode
+
+dataDecider :: Seconds -> NonEmpty (Meters, Seconds) -> (Meters, Seconds)
+dataDecider threshold (x :| xs) = foldl' decider x xs
+  where
+    decider (distance1, duration1) (distance2, duration2)
```

Comment:
Logic should be as follows
```
decider (distance1, duration1) (distance2, duration2) = do
    let durationDiff = abs $ duration1 - duration2

    -- If durationDiff is within threshold then choose shorter route
    if durationDiff <= threshold
        then if distance1 < distance2
            then (distance1, duration1)
            else (distance2, duration2)
        
        -- Otherwise choose faster route
        else if duration1 < duration2
            then (distance1, duration1)
            else (distance2, duration2)
```

Reply:
Also the decider function should be passed in from the application and not be in shared-kernel

Reply:
Done